### ISSUE TITLE
Visualize attention

### ISSUE THREAD

**Issue by huynhtruc0309** [yahoo/maaf#3]
Thank you for your work.
Can you share the code for visualizing attention?

**Comment by emdodds** [comment]
Thanks for your interest in our work! Unfortunately the visualization code would not be easy to share. Briefly, what we did is: (1) modify the Transformer code to save the attention maps to disk, (2) run the model with this modification for each of several inputs, loading the attention map from disk after running each example. (3) For the attention map for "sleeves" for example we subtracted a "baseline" attention map obtained with many random words in the place of "sleeves". 

Sorry I can't be of more help with this, at least on short notice.

**Comment by huynhtruc0309** [comment]
Dear author,
Do you give options to run the model basing on each method in this table?
![image](https://user-images.githubusercontent.com/37890714/118483376-22871480-b740-11eb-84fe-55ca0a95d454.png)
Thank you for your answer.

**Comment by emdodds** [comment]
Yes, you can try these variations using the `--attn_2stream_mode` parameter with `--model` set to `attention`. The default `xxx_xmm_xff` corresponds to the second line of this table. In general you can pass in a string that will be parsed into a sequence of attention operations [here](https://github.com/yahoo/maaf/blob/master/models/TransformerExtensions.py#L49).
